network/src/peers/peer_info.rs

```diff
@@ -91,8 +108,7 @@ pub struct PeerInfo {
     /// The bool indicates whether it's abortable - otherwise it
     /// needs to be awaited instead.
     #[serde(skip)]
-    #[allow(clippy::type_complexity)]
-    pub tasks: Arc<Mutex<Vec<(task::JoinHandle<()>, bool)>>>,
+    tasks: DropJoin<task::JoinHandle<()>>,
```

Comment:
we need to be able to `.await` the outbound connection tasks instead of just `.abort()ing` them, otherwise we will not be able to accurately measure the capacity of outbound channels over time - can we add such a functionality to `DropJoin`?

Reply:
Then we don't even need `DropJoin` for the tasks where this is applicable. If the futures are spawned in tokio tasks, then tokio will continue to poll them directly.

Reply:
Is the concern here a race over the metrics reporting after reading?

Reply:
The issue is that when outbound messages get queued (and bump the related metric showing how many messages are currently queued to be sent), but the corresponding `Receiver` doesn't `recv` them (e.g. when the owning task gets `abort()`ed abruptly), we are stuck with lingering entries that over time suggest that we constantly have outbound messages waiting to be sent, while in reality we are sending them immediately.